### DISCUSSION TITLE
Huawei Emma installation offered, whats the best?

### DISCUSSION THREAD

**Discussion by borgqueenx** [wlcrs/huawei_solar#869] [General]
My installation comes with a huawei emma, if i decide to take it. Im wondering if its possible to connect to it, or should i rather connect to the inverter itself if that works? The installation comes with the newer luna battery of 6.9kwh.

I have been reading that the inverter can reboot if it loses connection to huawei servers? Is there a way to block the device from the internet all together? I am not comfortable with permanently running on chinese servers.

I am confused from the wiki about modbus installation and sdongle, as it does not say anything about huawei emma.

**Comment by mithri** [comment]
Hi @borgqueenx, I got it working with the help of https://github.com/wlcrs/huawei_solar/issues/498. You connect to EMMA and via that connection you will get the infos from the inverter, too.
